File: dataset/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Task30.py

def longest(s1, s2):
    if s1 is None or s2 is None:
        return ""
    
    unique_chars = set()
    
    for c in s1:
        if 'a' <= c <= 'z':
            unique_chars.add(c)
    
    for c in s2:
        if 'a' <= c <= 'z':
            unique_chars.add(c)
    
    return ''.join(sorted(unique_chars))


if __name__ == "__main__":
    # Test case 1
    a1 = "xyaabbbccccdefww"
    b1 = "xxxxyyyyabklmopq"
    print(f"Test 1: {longest(a1, b1)}")
    
    # Test case 2
    a2 = "abcdefghijklmnopqrstuvwxyz"
    b2 = "abcdefghijklmnopqrstuvwxyz"
    print(f"Test 2: {longest(a2, b2)}")
    
    # Test case 3
    a3 = "aretheyhere"
    b3 = "yestheyarehere"
    print(f"Test 3: {longest(a3, b3)}")
    
    # Test case 4
    a4 = "loopingisfunbutdangerous"
    b4 = "lessdangerousthancoding"
    print(f"Test 4: {longest(a4, b4)}")
    
    # Test case 5
    a5 = ""
    b5 = "abc"
    print(f"Test 5: {longest(a5, b5)}")

File: dataset/Vanilla/GEMINI_gemini-2.5-pro/Task30.cpp
#include <iostream>
#include <string>
#include <set>
#include <algorithm>

class Task30 {
public:
    static std::string longest(const std::string& s1, const std::string& s2) {
        // A set automatically stores unique elements in sorted order.
        std::set<char> chars;
        
        // Insert all characters from s1 and s2 into the set.
        chars.insert(s1.begin(), s1.end());
        chars.insert(s2.begin(), s2.end());
        
        // Construct a new string from the characters in the set.
        return std::string(chars.begin(), chars.end());
    }
};

int main() {
    // Test case 1
    std::cout << "Test 1: " << Task30::longest("xyaabbbccccdefww", "xxxxyyyyabklmopq") << std::endl;

    // Test case 2
    std::cout << "Test 2: " << Task30::longest("abcdefghijklmnopqrstuvwxyz", "abcdefghijklmnopqrstuvwxyz") << std::endl;

    // Test case 3
    std::cout << "Test 3: " << Task30::longest("aretheyhere", "yestheyarehere") << std::endl;

    // Test case 4
    std::cout << "Test 4: " << Task30::longest("loopingisfunbutdangerous", "lessdangerousthancoding") << std::endl;

    // Test case 5
    std::cout << "Test 5: " << Task30::longest("inmanylanguages", "theresapairoffunctions") << std::endl;

    return 0;
}